### PR TITLE
EPS-19: Set up the project and create entities

### PR DESCRIPTION
**Pull request by vladqf**
- Set up database connection and HikariCP.
- Set up flyway migration.
- Created entities.
- Added several configurations (application-dev.yml, applicaion-prod.yml)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    jakartaServletApiVersion = '6.1.0'
+    lombokVersion = '1.18.34'
+    loggerVersion = '1.5.12'
+}
+
+dependencies {
+    implementation "org.yaml:snakeyaml:$snakeyamlVersion"
+    implementation "org.springframework:spring-core:$springCoreVersion"
+    implementation "org.springframework:spring-context:$springContextVersion"
+    implementation "org.springframework:spring-webmvc:$springWebMvcVersion"
```
Comment (PavelShynkarou) [inline build.gradle:31]: As far as I remember spring-mvc already include core and context dependencies. Could you please recheck it?

File: src/main/java/com/ifortex/internship/model/Course.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.math.BigDecimal;
+import java.time.LocalDateTime;
+import java.util.Set;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
+public class Course {
```
Comment (PavelShynkarou) [inline src/main/java/com/ifortex/internship/model/Course.java:16]: Does it necessary use Builder and All/NoArgsConstructors at the same time? From my perspective the better way to use just one approach.

File: src/main/java/com/ifortex/internship/model/Student.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+@Data
+@AllArgsConstructor
+@NoArgsConstructor
+@Builder
```
Comment (PavelShynkarou) [inline src/main/java/com/ifortex/internship/model/Student.java:11]: the same

**Review by PavelShynkarou — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 1 above)
Comment (vladqf) [reply build.gradle:27]: You're right. Removed spring core and spring context dependencies

File: src/main/java/com/ifortex/internship/model/Course.java (hunk 2 above)
Comment (vladqf) [reply src/main/java/com/ifortex/internship/model/Course.java:11]: Removed constructor annotations. I'll add them later if necessary

File: src/main/java/com/ifortex/internship/model/Student.java (hunk 3 above)
Comment (vladqf) [reply src/main/java/com/ifortex/internship/model/Student.java:7]: Also removed constructor annotations.

**Review by PavelShynkarou — APPROVED** [review]

**Pull request merged**